specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/dataVersion.json

```diff
@@ -0,0 +1,387 @@
+          },
+          {
+            "$ref": "#/parameters/DataVersionVersionsParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DataVersionResource"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "machineLearningServices.json#/definitions/MachineLearningServiceError"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "DataVersion"
+        ],
+        "x-ms-examples": {
+          "Create or update Data Version": {
+            "$ref": "./examples/createOrUpdateDataVersion.json"
+          }
+        },
+        "description": "Create or update Data Version.",
+        "operationId": "DataVersions_CreateOrUpdate",
+        "parameters": [
+          {
+            "$ref": "machineLearningServices.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/APIVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/DataVersionNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DataVersionVersionsParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Data Version entity to create or update.",
+            "schema": {
+              "$ref": "#/definitions/DataVersionResourceRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DataVersionResource"
+            }
+          },
+          "201": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DataVersionResource"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "machineLearningServices.json#/definitions/MachineLearningServiceError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "DataVersionNameParameter": {
+      "in": "path",
+      "name": "name",
+      "type": "string",
+      "description": "Data Version Name identifier.",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    },
+    "DataVersionVersionsParameter": {
+      "in": "path",
+      "name": "version",
+      "type": "string",
+      "description": "Data Version Version identifier.",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    }
+  },
+  "definitions": {
+    "DataVersionResource": {
```

Comment:
See comments on adding a type for the resource envelope as above.  It would be best to have a single resource you can reference from all of these